8135292: Remove duplicate code in <OS>Address.java in SA

jhuttana opened the pull request:
This PR is to address : https://bugs.openjdk.org/browse/JDK-8135292?jql=labels%20%3D%20starter-bug
Verified the build before and after the patch. Also below tests are run:
Before Patch:
```
$ make test TEST="jtreg:test/hotspot/jtreg/serviceability"
	==============================
	Test summary
	==============================
	   TEST                                              TOTAL  PASS  FAIL ERROR   
	>> jtreg:test/hotspot/jtreg/serviceability             316   314     1     1 <<
	==============================
	TEST FAILURE
```

After Patch:
```
$ make test TEST="jtreg:test/hotspot/jtreg/serviceability"
	==============================
	Test summary
	==============================
	   TEST                                              TOTAL  PASS  FAIL ERROR   
	>> jtreg:test/hotspot/jtreg/serviceability             316   314     1     1 <<
	==============================
	TEST FAILURE
```

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8135292](https://bugs.openjdk.org/browse/JDK-8135292): Remove duplicate code in <OS>Address.java in SA


### Reviewers
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**) ⚠️ Review applies to [c9da1be2](https://git.openjdk.org/jdk/pull/9112/files/c9da1be2028bd18902009ab03f197686fabd55b7)
 * [Alex Menkov](https://openjdk.org/census#amenkov) (@alexmenkov - **Reviewer**) ⚠️ Review applies to [c9da1be2](https://git.openjdk.org/jdk/pull/9112/files/c9da1be2028bd18902009ab03f197686fabd55b7)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9112/head:pull/9112` \
`$ git checkout pull/9112`

Update a local copy of the PR: \
`$ git checkout pull/9112` \
`$ git pull https://git.openjdk.org/jdk pull/9112/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9112`

View PR using the GUI difftool: \
`$ git pr show -t 9112`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9112.diff">https://git.openjdk.org/jdk/pull/9112.diff</a>

</details>

jhuttana commented:
/covered

jhuttana commented:
@jerboaa Thank you for guiding me on what was needed further.
Now I have completed the suggested steps. 
I didn't get much about what is jcheck !

jerboaa commented:
> I didn't get much about what is jcheck !

When you click on the `Details` link you'll see:
```
 - OCA signatory status must be verified
 - Too few reviewers with at least role reviewer found (have 0, need at least 1)
```

Think of it as a tool that performs basic checks. Once all checks are in order, a `ready` label will be added and you'll be able to '/integrate' with a sponsor. The bots will guide you through this. On step at a time.

jhuttana commented:
Ok! Thank you!

TheShermanTanker commented:
@jhuttana Try not to force push when contributing to OpenJDK, unless there isn't a better solution, doing so makes it harder to review the PR

jhuttana commented:
> @jhuttana Try not to force push when contributing to OpenJDK, unless there isn't a better solution, doing so makes it harder to review the PR

Sure I will take care of that next time. 
As I am new to this process, now I have a confusion :) i.e, how to handle the scenarios where if something needs to be changed in the commit message or changes in any files that I realize  after pushing and raising a PR ?

jerboaa commented:
> how to handle the scenarios where if something needs to be changed in the commit message or changes in any files that I realize after pushing and raising a PR ?

You don't need to change the commit messages. Once reviewed, the commit message will be set by the bots. Just look at the git log of existing commits. All you can do to influence the message is using `/summary` or the likes. In a nutshell, don't do it. If you need to add extra changes. Add them as separate commits on top of the previous one to your PR branch. That's it. The bot will rebase and merge commits when integrating.

plummercj commented on src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/debugger/bsd/BsdDebuggerLocal.java:
```diff
@@ -525,7 +525,7 @@ public long readJLong(long address)
     public long getAddressValue(Address addr) {
       if (addr == null) return 0;
-      return ((BsdAddress) addr).getValue();
+      return ((BsdAddress) addr).asLongValue();
```
`asLongValue()` is declared in `Address`, so it should not be necessary to cast to a `BsdAddress`

jhuttana replied:
ok sure

plummercj commented:
There are also SA tests in `test/jdk/sun/tools/jhsdb` that should be run.

jhuttana commented:
> There are also SA tests in `test/jdk/sun/tools/jhsdb` that should be run.

I will run those tests too once after making the changes you suggested.

jhuttana commented:
Oops! previously I was not know that I should not force push and should not edit commit message so this time I took care of those points. But, rebase is also not suggested :(
I did so because after making suggested review changes, when I tried,
```
$ git add <changed files>
$ git commit --amend --no-edit
$ git push <remote-to-my-fork> <PR_Branch> 
To https://github.com/jhuttana/jdk.git
 ! [rejected]                JDK-8135292 -> JDK-8135292 (non-fast-forward)
error: failed to push some refs to 'https://github.com/jhuttana/jdk.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```
So I thought I should git pull. 
```
# git pull
hint: You have divergent branches and need to specify how to reconcile them.
hint: You can do so by running one of the following commands sometime before
hint: your next pull:
hint: 
hint:   git config pull.rebase false  # merge
hint:   git config pull.rebase true   # rebase
hint:   git config pull.ff only       # fast-forward only
hint: 
hint: You can replace "git config" with "git config --global" to set a default
hint: preference for all repositories. You can also pass --rebase, --no-rebase,
hint: or --ff-only on the command line to override the configured default per
hint: invocation.

# git config pull.rebase true
# git pull
Auto-merging src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/debugger/bsd/BsdDebuggerLocal.java
CONFLICT (content): Merge conflict in src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/debugger/bsd/BsdDebuggerLocal.java
Auto-merging src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/debugger/dummy/DummyDebugger.java
CONFLICT (content): Merge conflict in src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/debugger/dummy/DummyDebugger.java
Auto-merging src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/debugger/linux/LinuxDebuggerLocal.java
CONFLICT (content): Merge conflict in src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/debugger/linux/LinuxDebuggerLocal.java
Auto-merging src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/debugger/remote/RemoteDebuggerClient.java
CONFLICT (content): Merge conflict in src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/debugger/remote/RemoteDebuggerClient.java
Auto-merging src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/debugger/windbg/WindbgDebuggerLocal.java
CONFLICT (content): Merge conflict in src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/debugger/windbg/WindbgDebuggerLocal.java
error: could not apply da58aea47c8... 8135291: Removed duplicate code in <OS>Address.java and related files in SA
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply da58aea47c8... 8135291: Removed duplicate code in <OS>Address.java and related files in SA -After asLongValue() added to interface Address.java, there no need to have the getValue() in every instance Address.java

# git add src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/debugger/bsd/BsdDebuggerLocal.java src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/debugger/dummy/DummyDebugger.java src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/debugger/linux/LinuxDebuggerLocal.java src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/debugger/remote/RemoteDebuggerClient.java src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/debugger/windbg/WindbgDebuggerLocal.java

# git rebase --continue
[detached HEAD 7af119a55b8] 8135291: Removed duplicate code in <OS>Address.java and related files in SA -After asLongValue() added to interface Address.java, there no need to have the getValue() in every instance Address.java
 5 files changed, 5 insertions(+), 5 deletions(-)
Successfully rebased and updated refs/heads/JDK-8135292.

# git push upstream JDK-8135292
Username for 'https://github.com': jhuttana
Password for 'https://jhuttana@github.com': 
Enumerating objects: 37, done.
Counting objects: 100% (37/37), done.
Delta compression using up to 16 threads
Compressing objects: 100% (18/18), done.
Writing objects: 100% (20/20), 9.66 KiB | 3.22 MiB/s, done.
Total 20 (delta 15), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (15/15), completed with 14 local objects.
To https://github.com/jhuttana/jdk.git
   8c61d54f1e3..7af119a55b8  JDK-8135292 -> JDK-8135292

```

Could you please let me know where did I went wrong?

jhuttana commented:
Further, after making the suggested changes I have run `test/jdk/sun/tools/jhsdb`
```
# make test TEST="jtreg:test/jdk/sun/tools/jhsdb"
		==============================
		Test summary
		==============================
		   TEST                                              TOTAL  PASS  FAIL ERROR   
		   jtreg:test/jdk/sun/tools/jhsdb                        7     7     0     0   
		==============================
		TEST SUCCESS
```

jerboaa commented:
> ```
> $ git add <changed files>
> $ git commit --amend --no-edit
>```

The `git commit --amend` changes the current commit. Don't use `--amend` and it should be fine.

jhuttana commented:
> > > ```
> > > $ git add <changed files>
> > > $ git commit --amend --no-edit
> > > ```
> > 
> > 
> > The `git commit --amend` changes the current commit. Don't use `--amend` and it should be fine.
> 
> Ok thank you.

I tried just : 
```
# git commit
Aborting commit due to empty commit message.
```
The commit is aborted.

plummercj reviewed (changes requested):
The changes look fine now, but. you will need to update the copyright in each file to 2022.

jhuttana commented:
/integrate

jerboaa commented:
/sponsor
